cloudsync/tasks.py

```diff
@@ -352,3 +352,37 @@ def parse_content_metadata(response):
         content_length = int(content_length)
 
     return file_name, content_type, content_length
+
+
+@shared_task(bind=True)
+def sort_transcoded_m3u8_files(self):
+    """
+    Sort files with highest resolution first in trancoded video playlist
+    """
+    for video in Video.objects.all():
+        for transcoded_video in video.transcoded_videos:
+            s3_filename = transcoded_video.s3_object_key
+            s3_client = boto3.client('s3')
+            file = s3_client.get_object(Bucket=settings.VIDEO_S3_TRANSCODE_BUCKET, Key=s3_filename)
+            file_content = file['Body'].read().decode()
+
+            delimiter = '#EXT-X-STREAM-INF:'
+            lines = file_content.split(delimiter)
+            header = lines.pop(0)
+
+            if str.strip(header) != "#EXTM3U":
+                continue
+
+            try:
+                lines.sort(key=lambda line: -int(re.search(r'RESOLUTION=(\d+)', line).group(1)))
```

Comment:
Might also need to sort by bandwidth, there are two codecs with the same resolution but one has a lower bandwidth than the other:
```
#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=471000,RESOLUTION=320x176,CODECS="avc1.42001e,mp4a.40.2"
video_1351620000001-200050.m3u8
#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=682000,RESOLUTION=320x176,CODECS="avc1.42001e,mp4a.40.2"
video_1351620000001-200040.m3u8
```

After reordering, the lower bandwidth codec precedes the higher bandwidth code in some videos but not others